### PR TITLE
HTTP protocol refactoring

### PR DESCRIPTION
**Pull request by Kriechi**
This PR starts of a larger refactoring process to unify everything related to HTTP semantics and split the actual "on-the-wire" protocol from the overall flow behind it.

Mostly moving code around between projects, modules, and classes.

### CODE REVIEW

**Comment by mhils** [comment]
If we create a subfolder for websockets, we should probably do the same for http?

Apart from that, LGTM. :-)

File: netlib/http_semantics.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Response(object):
```
Comment (mhils) [inline netlib/http_semantics.py]: What's wrong with namedtuple?

Comment (Kriechi) [inline netlib/http_semantics.py]: it was already a class somewhere - and somewhere else it was used as namedtuple...
I decided a proper class is probably cleaner.

Comment (mhils) [inline netlib/http_semantics.py]: Ok - I reviewed the netlib PR first which is why I missed pathod's Response class here.

**Comment by mhils** [comment]
One more point: If we create a subfolder for websockets, we should probably do so for http as well (there are five `http_` files if I counted correctly), but that can be done in a separate PR.

**Comment by Kriechi** [comment]
I updated the changes with a new file & module structure.

**Pull request merged**